### PR TITLE
Update the build script for Go

### PR DESCRIPTION
**Pull request by easwars**
The version of Go available from apt-get is Go1.11.x, which is really
old, and the xds implementation in gRPC-Go uses features not supported
in Go1.11. So, we need to be using the latest version of Go.

Summary of changes
* Pull the latest version of Go from the downloads page
* Extract the Go runtime/toolchain into /usr/local
* Copy the `go` command line binary into /usr/bin so that it is
  available in $PATH
* Do a `go get` on the gRPC-Go module to pull in the latest version

### CODE REVIEW

**Comment by easwars** [comment]
@menghanl

File: scripts/create_service.sh [hunk 1]
```diff
@@ -26,8 +26,13 @@ fi
...
     go)
-        build_script="cd \"traffic-director-grpc-examples/go/${service_type}_server\"
-sudo apt-get install -y golang
+        build_script="sudo apt-get install -y wget
+wget https://dl.google.com/go/go1.16.5.linux-amd64.tar.gz
+sudo tar -C /usr/local -xvf go1.16.5.linux-amd64.tar.gz
+sudo cp /usr/local/go/bin/go /usr/bin/go
+cd \"traffic-director-grpc-examples/go/${service_type}_server\"
+go build .
+go get google.golang.org/grpc
```
Comment (menghanl) [inline scripts/create_service.sh:35]: Why is this necessary?
The `go build .` above should download the dependencies based on `go.mod`.

File: scripts/create_service.sh [hunk 2]
```diff
@@ -26,8 +26,13 @@ fi
 
 case "${language}" in
     go)
-        build_script="cd \"traffic-director-grpc-examples/go/${service_type}_server\"
-sudo apt-get install -y golang
+        build_script="sudo apt-get install -y wget
+wget https://dl.google.com/go/go1.16.5.linux-amd64.tar.gz
+sudo tar -C /usr/local -xvf go1.16.5.linux-amd64.tar.gz
```
Comment (menghanl) [inline scripts/create_service.sh:31]: What would tar do if `/usr/local/go` already exists?
Maybe we can install this go to `$HOME`, then update `PATH`

File: scripts/create_service.sh (hunk 1 above)
Comment (easwars) [reply scripts/create_service.sh:35]: Without this, we would have to update this `go.mod` whenever we have a new gRPC version. Is that what we want?

File: scripts/create_service.sh (hunk 2 above)
Comment (easwars) [reply scripts/create_service.sh:31]: The reason I found doing this useful is that if something went bad and you have to SSH into the VM to check things out, any updates done to `PATH` by the startup script are not reflected. So, putting `go` in `/usr/bin` which you are sure is part of `PATH` ensures that `go` command is available.

Comment (easwars) [reply scripts/create_service.sh:31]: > What would tar do if /usr/local/go already exists?

I could use one of these options:
```
       --overwrite
              Overwrite existing files when extracting.

       --overwrite-dir
              Overwrite metadata of existing directories when extracting
              (default).
```

File: scripts/create_service.sh (hunk 1 above)
Comment (menghanl) [reply scripts/create_service.sh:35]: > update this go.mod whenever we have a new gRPC version

I think that is the right thing to do.

**Review by menghanl — APPROVED** [review]
